### PR TITLE
Migrate to mini-css-extract-plugin from extract-text-webpack-plugin

### PR DESCRIPTION
**Pull request by sibiraj-s**
**Migrated from `ETWP` to `mini-css-extract-plugin`**. Webpack discourages the use of ETWP since webpack 4.0

Since `mini-css-extract-plugin` requires webpack>4.4, hence webpack is updated.
Which inturn causes conflicts with `contenthash` so to fix tests. I renamed it to **`templatehash`**.. I can't think of more suitable name than this.

**Breaking**: `contenthash` is renamed to `templatehash`

### CODE REVIEW

**Comment by jantimon** [comment]
Sorry this pr includes to many features and also some quite opinionated ones

**Comment by sibiraj-s** [comment]
@jantimon there are no new features added this is more like a cleanup. I barely renamed few stuffs.

**Comment by sibiraj-s** [comment]
What about migrating to `mini-css-extract-plugin`?

**Comment by edmorley** [comment]
In general it's best to keep to one type of change per PR. Looking at the changes here, I see linting fixes, gitignore changes, test cleanups, dependency bumps, deleting dist folders and the switch to mini-css-extract-plugin. Each of those should likely be a separate PR - which then makes it simpler to review/discuss :-)

**Comment by sibiraj-s** [comment]
Yes @edmorley My bad. I thought these changes will be good. but separate PR.

@jantimon can you reopen the PR. I will update this to the change with shifting from 

ETWP ->mini-css-extract-plugin

**Comment by sibiraj-s** [comment]
@jantimon @edmorley Updated the changes. This includes `dist/` folder which is auto-generated

File: index.js [hunk 1]
```diff
@@ -279,10 +279,10 @@ class HtmlWebpackPlugin {
               return self.options.showErrors ? prettyError(err, compiler.context).toHtml() : 'ERROR';
             })
             .then(html => {
-              // Allow to use [contenthash] as placeholder for the html-webpack-plugin name
+              // Allow to use [templatehash] as placeholder for the html-webpack-plugin name
```
Comment (edmorley) [inline index.js]: This change seems unrelated?

File: examples/custom-template/webpack.config.js [hunk 2]
```diff
@@ -13,14 +13,14 @@ module.exports = {
...
+      { test: /\.css$/, use: [MiniCssExtractPlugin.loader, 'css-loader'] },
       { test: /\.png$/, loader: 'file-loader' }
     ]
   },
   plugins: [
     new HtmlWebpackPlugin({
       template: 'template.html'
     }),
-    new ExtractTextPlugin('styles.css')
+    new MiniCssExtractPlugin({ filename: 'style.css' })
```
[the remark this answers is not in the document]
Comment (sibiraj-s) [reply examples/custom-template/webpack.config.js]: yes, too many diff. I missed that. Updated now

Checkout #1064

File: index.js (hunk 1 above)
Comment (sibiraj-s) [reply index.js]: That is unrelated yes. But tests will fail without that.

Comment (edmorley) [reply index.js]: Ah makes sense. I guess this change fixes #1033.

Comment (sibiraj-s) [reply index.js]: Yes. Just noticed there is also another PR #1034 does the same.

**Comment by sibiraj-s** [comment]
What about updating webpack in peerDependency to `>= 4.4` as mini-css plugin. these plugins will be used together mostly

File: index.js (hunk 1 above)
Comment (jantimon) [reply index.js]: Actually it's fine for me - maybe we could also replace `contenthash` to `templatehash` before we compile it.  
So users will be able to use `contenthash` and `templatehash`

**Comment by jantimon** [comment]
Thanks alot 👍

File: index.js (hunk 1 above)
Comment (sibiraj-s) [reply index.js]: Cool. So that no breaking change will be introduced. 👍

Comment (sibiraj-s) [reply index.js]: @jantimon checkout #1065
